File: bolao_de_computacao/lib/app/model/championship.dart
import 'package:bolao_de_computacao/app/model/round.dart';

class Championship{

  String name;
  int currentRound;
  int roundAmount;
  List<Round> rounds;

  Championship(this.name, this.currentRound, this.roundAmount, this.rounds);

  String getTitle(){
    return "$name - ${this.rounds[currentRound].getTitle()}";
  }
}

File: bolao_api/src/main/kotlin/com/capivara/bolao/model/Championship.kt
package com.capivara.bolao.model

import com.fasterxml.jackson.annotation.JsonProperty
import java.util.*

class Championship(@JsonProperty("id")
                   var id: UUID?,
                   @JsonProperty("name")
                   val name: String,
                   @JsonProperty("season")
                   val season: String,
                   @JsonProperty("current_round")
                   val currentRound: Int,
                   @JsonProperty("type")
                   val type: Int,
                   @JsonProperty("title_image_url")
                   val titleImageUrl: String,
                   @JsonProperty("round_list")
                   val roundList: List<Round>
)